### ISSUE TITLE
i want to redefine the  "Validation Error"

### ISSUE THREAD

**Issue by tonglin96** [tiangolo/fastapi#553]
i want to change the status code and response body for "validation error",what should i do ?

**Comment by dmontagu** [comment]
I haven't confirmed this works, but you should be able to do:

```python
async def custom_request_validation_exception_handler(
    request: Request, exc: RequestValidationError
) -> JSONResponse:
    # change this as you like, including possibly the return type
    return JSONResponse(
        status_code=HTTP_422_UNPROCESSABLE_ENTITY, content={"detail": exc.errors()}
    )

...

app = FastAPI(...)

...

app.add_exception_handler(
    fastapi.exceptions.RequestValidationException,
    custom_request_validation_exception_handler
)
```

**Comment by a372661369** [comment]
> ```python
> app.add_exception_handler(
>     fastapi.exceptions.RequestValidationException,
>     custom_request_validation_exception_handler
> )
> ```
but  i want this exception shows in doc,it looks like must change the source code

**Comment by a372661369** [comment]
it looks like i cant cover the "RequestValidationError", so i create a new exception like this:
`class ValidException(Exception):
    def __init__(self, code, msg):
        self.code = code
        self.msg = msg


async def custom_request_validation_exception_handler(
    request: Request, exc: ValidException
) -> JSONResponse:
    # change this as you like, including possibly the return type
    return JSONResponse(
        status_code=200, content={"code": exc.code, "msg": exc.msg}
    )

app.add_exception_handler(
    ValidException,
    custom_request_validation_exception_handler
)
`

**Comment by tiangolo** [comment]
Yep, as @dmontagu says. It's documented here: https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions
